cpp/src/buffer/buffer.cpp

```diff
@@ -73,30 +96,55 @@ std::unique_ptr<Buffer> Buffer::copy(MemoryType target, rmm::cuda_stream_view st
         return copy(stream);
     }
 
+    auto event = std::make_shared<Event>();
     return std::visit(
         overloaded{
             [&](const HostStorageT& storage) -> std::unique_ptr<Buffer> {
-                return std::unique_ptr<Buffer>(new Buffer{
-                    std::make_unique<rmm::device_buffer>(
-                        storage->data(), storage->size(), stream, br->device_mr()
-                    ),
-                    br
-                });
+                // Have to ensure that any async work is complete.
+                synchronize();
```

Comment:
Why?

Reply:
This one was wrong, so I fixed it with a stream dependency.